File: data/derived_cot/rq3_prompting/results/translation/qwen/py_to_cpp/code_output/RPGCharacter.cpp
#include <string>

class RPGCharacter {
private:
    std::string name;
    int hp;
    int attack_power;
    int defense;
    int level;
    int exp;

public:
    RPGCharacter(const std::string& name, int hp, int attack_power, int defense, int level = 1)
        : name(name), hp(hp), attack_power(attack_power), defense(defense), level(level), exp(0) {}

    void attack(RPGCharacter& other_character) {
        int damage = std::max(attack_power - other_character.defense, 1);
        other_character.hp -= damage;
    }

    int heal() {
        hp += 10;
        if (hp > 100) {
            hp = 100;
        }
        return hp;
    }

    void gain_exp(int amount) {
        while (amount != 0) {
            if (exp + amount >= level * 100) {
                amount -= (level * 100 - exp);
                level_up();
            } else {
                exp += amount;
                amount = 0;
            }
        }
    }

    void level_up() {
        if (level < 100) {
            level++;
            exp = 0;
            hp += 20;
            attack_power += 5;
            defense += 5;
        }
    }

    bool is_alive() const {
        return hp > 0;
    }
};

File: data/derived_cot/rq2_eval/translation/r1/_converted_code/java_to_py/RPGCharacter.py
class RPGCharacter:

    def __init__(self, name, hp, attack_power, defense, level=1):
        self.name = name
        self.hp = hp
        self.attack_power = attack_power
        self.defense = defense
        self.level = level
        self.exp = 0

    def attack(self, other_character):
        damage = max(self.attack_power - other_character.defense, 1)
        other_character.hp -= damage

    def heal(self):
        self.hp += 10
        if self.hp > 100:
            self.hp = 100
        return self.hp

    def gain_exp(self, amount):
        while amount != 0:
            required = self.level * 100 - self.exp
            if amount >= required:
                amount -= required
                self.level_up()
            else:
                self.exp += amount
                amount = 0

    def level_up(self):
        if self.level < 100:
            self.level += 1
            self.exp = 0
            self.hp += 20
            self.attack_power += 5
            self.defense += 5
        return [self.level, self.hp, self.attack_power, self.defense]

    def is_alive(self):
        return self.hp > 0

    def getName(self):
        return self.name

    def getHp(self):
        return self.hp

    def getAttackPower(self):
        return self.attack_power

    def getDefense(self):
        return self.defense

    def getLevel(self):
        return self.level

    def getExp(self):
        return self.exp